tests/python/pants_test/build_graph/test_target.py

```diff
@@ -27,6 +30,20 @@ class ImplicitSourcesTestingTargetMulti(Target):
   default_sources_exclude_globs = ('*.baz', '*.qux')
 
 
+class SourcesTarget(Target):
```

Comment:
This could extend `Files` and just pass `sources` up to its constructor.

Reply:
I tried this, but it fails another test due to the overloading that `Files` targets do for `has_sources`.